glommio/src/sys/uring.rs

```diff
@@ -1512,10 +1513,24 @@ impl Reactor {
         match latency {
             None => Self::simple_poll(&self.poll_ring, woke),
             Some(Latency::NotImportant) => Self::simple_poll(&self.main_ring, woke),
-            Some(Latency::Matters(_)) => Self::simple_poll(&self.latency_ring, woke),
+            Some(Latency::Matters(_)) => {
+                let ret = Self::simple_poll(&self.latency_ring, woke);
+                if let Some(src) = &*self.timeout_src.borrow() {
+                    if src.result().is_some() {
+                        // the preemption timer has fired during the last poll
+                        // we need to communicate this to the scheduler
+                        let _ = executor().yield_task_queue_now();
```

Comment:
I don't understand why this is being tested only in the `Latency::Matters` case. 
Even if you don't have a latency requirement, other task queues might, so if the timer fired you have to bail.

am I missing anything ?

Reply:
Yeah, the preemption timer is added to the latency ring so if we rush to any other ring, that's not an issue.